src/main/java/com/ibanity/apis/client/http/impl/IbanityHttpClientImpl.java

```diff
@@ -0,0 +1,113 @@
+package com.ibanity.apis.client.http.impl;
+
+import com.ibanity.apis.client.http.IbanityHttpClient;
+import com.ibanity.apis.client.http.handler.IbanityResponseHandler;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.message.BasicHeader;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static com.ibanity.apis.client.http.IbanityHttpUtils.objectMapper;
+import static org.apache.http.HttpHeaders.AUTHORIZATION;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+
+public class IbanityHttpClientImpl implements IbanityHttpClient {
+
+    private final HttpClient httpClient;
+    private final IbanityResponseHandler ibanityResponseHandler;
+
+    public IbanityHttpClientImpl(HttpClient httpClient) {
+        this.httpClient = httpClient;
+        ibanityResponseHandler = new IbanityResponseHandler();
+    }
+
+    @Override
+    public String get(URI path) {
+        return get(path, null, newHashMap());
+    }
+
+    @Override
+    public String get(URI path, String customerAccessToken) {
+        return get(path, customerAccessToken, newHashMap());
+    }
+
+    @Override
+    public String get(URI path, String customerAccessToken, Map<String, String> additionalHeaders) {
+        try {
+            HttpGet httpGet = new HttpGet(path);
+            addHeaders(customerAccessToken, additionalHeaders, httpGet);
+            return httpClient.execute(httpGet, ibanityResponseHandler);
+        } catch (IOException exception) {
+            throw new RuntimeException("An error occurred while connecting to Ibanity", exception);
+        }
+    }
+
+    @Override
+    public String post(URI path, Object requestApiModel) {
+        return post(path, requestApiModel, null, newHashMap());
+    }
+
+    @Override
+    public String post(URI path, Object requestApiModel, String customerAccessToken) {
+        return post(path, requestApiModel, customerAccessToken, newHashMap());
```

Comment:
The goal is to avoid handling null everywhere but instead play with empty collections.
It reduces the risk of NPE and improve code readability by removing the need for null checks and allowing fluent method calls.